example_docs/docs/conf.py

```diff
@@ -100,3 +101,61 @@
     # Default image for thumbnails.
     "**": "_static/images/logo.png",
 }
+
+
+def linkcode_resolve(domain, info):
+    """
+    Determine the URL corresponding to Python object
+    """
+    if domain != "py":
+        return None
+
+    modname = info["module"]
+    fullname = info["fullname"]
+    # print("Mod name: {}".format(modname))
+    # print("Full name: {}".format(fullname))
+
+    submod = sys.modules.get(modname)
+    # print("Submod: {}".format(submod))
+    if submod is None:
+        return None
+
+    obj = submod
+    for part in fullname.split("."):
+        # print("Part: {}".format(part))
+        obj = getattr(obj, part)
+        # print("Obj: {}".format(obj))
+        if not (inspect.isclass(obj) or inspect.ismethod(obj) or inspect.isfunction(obj)):
+            return None
+
+    try:
+        fn = inspect.getsourcefile(obj).split("qiskit_sphinx_theme")[1]
+    except Exception as e:
+        # print("Error: {}".format(e))
+        return None
+    # print("Fn: {}".format(fn))
+
+    try:
+        source, lineno = inspect.getsourcelines(obj)
+        # print("lineno: {}".format(lineno))
+    except Exception:
+        lineno = None
+    # print("Source: {}".format(source))
+
+    if lineno:
+        linespec = "#L%d-L%d" % (lineno, lineno + len(source) - 1)
+    else:
+        linespec = ""
+    # print("linespec: {}".format(linespec))
+
+    #
+    # if 'dev' in release:
+    #     return "https://github.com/Qiskit/documentation/%s%s" % (
+    #        fn, linespec)
+    # else:
+    #     return "https://github.com/numpy/numpy/blob/v%s/numpy/%s%s" % (
+    #        release, fn, linespec)
+
+    # print("Returning https://github.com/Qiskit/qiskit_sphinx_theme/tree/main{}/{}".format(fn, linespec))
+
+    return "https://github.com/Qiskit/qiskit_sphinx_theme/tree/main{}/{}".format(fn, linespec)
```

Comment:
It looks like this isn't working fully. If you look at the file `example_docs/docs/_build/stubs/api_example.Electron.html` in Sphinx's output, it has URLs like this:

https://github.com/Qiskit/qiskit_sphinx_theme/tree/main//#L7-L98

https://github.com/Qiskit/qiskit_sphinx_theme/tree/main//#L46-L57

https://github.com/Qiskit/qiskit_sphinx_theme/tree/main//#L59-L62


They should be linking to the file https://github.com/Qiskit/qiskit_sphinx_theme/blob/main/example_docs/api_example/electron.py. It's fine to hardcode  `blob/main/example_docs` in the URL, but you should be dynamically computing the `api_example/electron.py` part because we will need similar logic for Qiskit.

The line numbers are working correclty though. That's so cool! Nice.

Reply:
I'm a little confused. Aren't the links being sent to https://github.com/Qiskit/qiskit_sphinx_theme/blob/main/example_docs/api_example/electron.py for the Electron attributes?